src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java

```diff
@@ -0,0 +1,176 @@
+/***********************************************************************
+ *                                                                     *
+ * Copyright Ericsson AB 2017                                          *
+ *                                                                     *
+ * No part of this software may be reproduced in any form without the  *
+ * written permission of the copyright owner.                          *
+ *                                                                     *
+ ***********************************************************************/
+package com.ericsson.ei.queryservice;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * This class is responsible to fetch the criterias from both the query
+ * parameters or the form parameters. Then find the aggregatedObject from the
+ * database and concatenate the result.
+ *
+ * @author xjibbal
+ *
+ */
+
+@Component
+public class ProcessQueryParams {
+
+	@Value("${aggregationCollectionName}")
+	private String aggregationCollectionName;
+
+	@Value("${aggregationDataBaseName}")
+	private String aggregationDataBaseName;
+
+	@Value("${missedNotificationCollectionName}")
+	private String missedNotificationCollectionName;
+
+	@Value("${missedNotificationDataBaseName}")
+	private String missedNotificationDataBaseName;
+
+	static Logger log = (Logger) LoggerFactory.getLogger(QueryServiceRestController.class);
+
+	@Autowired
+	ProcessAggregatedObject processAggregatedObject;
+
+	@Autowired
+	ProcessMissedNotification processMissedNotification;
+
+	/**
+	 * This method takes the parameters from the REST POST request body and
+	 * process it to create a JsonNode request to query the Aggregated Objects.
+	 * If the Aggregated Object matches the condition, then it is returned.
+	 *
+	 * @param request
+	 * @return JSONArray
+	 * @throws JsonProcessingException
+	 * @throws IOException
+	 */
+	public JSONArray filterFormParam(JsonNode request) throws JsonProcessingException, IOException {
+		JsonNode criteria = request.get("criteria");
+		JsonNode options = request.get("options");
+		log.info("The criteria is : " + criteria.toString());
+		log.info("The options is : " + options.toString());
+		JSONArray resultAggregatedObject = null;
+		JSONArray resultMissedNotification = null;
+		if (options.toString().equals("{}") || options.isNull()) {
```

Comment:
strings which are to be checked have to be the first ones, ex: if("".equals(obj))